pkg/action/executor/gitops/github_manager.go

```diff
@@ -353,60 +356,131 @@ func (g *GitHandler) getTree(ref *github.Reference, fileContent []byte) (tree *g
 		},
 	}
 
+	// If both a tree and a nested path modifying that tree are specified, this endpoint
+	// will overwrite the contents of the tree with the new path contents, and create
+	// a new tree structure, in effect causing an update tree semantics.
+
+	// ref.Object.SHA
+	// The SHA1 of an existing Git tree object which will be used as the base for the new tree.
+	// If provided, a new Git tree object will be created from entries in the Git tree object
+	// pointed to by base_tree and entries defined in the tree parameter. Entries defined in the
+	// tree parameter will overwrite items from base_tree with the same path. If you're creating
+	// new changes on a branch, then normally you'd set base_tree to the SHA1 of the Git tree
+	// object of the current latest commit on the branch you're working on. If not provided,
+	// GitHub will create a new Git tree object from only the entries defined in the tree
+	// parameter. If you create a new commit pointing to such a tree, then all files which
+	// were a part of the parent commit's tree and were not defined in the tree parameter
+	// will be listed as deleted by the new commit.
 	tree, _, err = g.client.Git.CreateTree(g.ctx, g.user, g.repo, *ref.Object.SHA, entries)
 	return tree, err
 }
 
-func (g *GitHandler) updateRemote(resName string, patches []PatchItem) (interface{}, error) {
-	yamlContent, err := g.getRemoteFileContent(resName, g.path)
-	if err != nil {
-		return nil, err
-	}
-
-	patchedYamlContent, err := ApplyPatch([]byte(yamlContent), patches)
-	if err != nil {
-		return nil, fmt.Errorf("error applying patches to file %s: patches: %v, err: %v", g.path, patches, err)
-	}
-
-	var branchRef *github.Reference
-	newBranchName := g.baseBranch + "-" + strconv.FormatInt(time.Now().UnixNano(), 32)
+func (g *GitHandler) updateRemote(resName, resNamespace, clusterId string, patches []PatchItem) (interface{}, error) {
+	var commitBranchName string
+	var commitBranchRef *github.Reference
+	var pr *github.PullRequest
+	var err error
 	if g.commitMode == commitModePR {
-		branchRef, err = g.createNewBranch(newBranchName)
+		pr, err = g.checkPrExists(resName, resNamespace, clusterId)
 		if err != nil {
-			return nil, fmt.Errorf("error creating new branch %s, %v", newBranchName, err)
+			return nil, err
+		}
+		if pr != nil {
+			glog.V(3).Infof("Found an existing PR #%d while executing action for "+
```

Comment:
nit: Maybe use V(2) here, which is the default. I think this is a proper message at v2 level for better troubleshooting if needed.